flink-core/src/main/java/org/apache/flink/api/common/io/compression/InflaterInputStreamFactory.java

```diff
@@ -23,13 +23,12 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Collection;
-import java.util.zip.InflaterInputStream;
 
 /**
  * Creates a new instance of a certain subclass of {@link java.util.zip.InflaterInputStream}.
  */
 @Internal
-public interface InflaterInputStreamFactory<T extends InflaterInputStream> {
+public interface InflaterInputStreamFactory<T extends InputStream> {
```

Comment:
I think Greg raised a good point, the classname and comments should be adjusted as well. How about `DecompressingStreamFactory`, or `DecoratorStreamFactory` (if you want to keep it generic)?